app/javascript/react/lib/handleReferentAssignation.js

```diff
@@ -0,0 +1,14 @@
+import Swal from "sweetalert2";
+import assignReferent from "../actions/assignReferent";
+
+const handleReferentAssignation = async (user, options = { raiseError: true }) => {
+  const result = await assignReferent(user.id, user.referentEmail);
+  if (result.success) {
+    user.updateWith(result.user);
```

Comment:
Je sais qu'on le fait à d'autres endroits mais c'est pas très logique qu'on fasse l'update ici plutôt que dans la classe user.
Je trouve pas ça bloquant mais sur le principe on devrait même pas ici prendre en entrée un user, mais juste un ID et un email, et retourner le résultat. 
C'est la classe user qui devrait se charger d'updater `this` en fonction du retour du call http

Reply:
yes ce serait sans doute mieux. Mais effectivement j'ai calqué sur ce qu'on faisait dans les autres méthodes `handle`, il faudrait faire un ticket/PR séparé pour refacto ça